Docs: Add AI section (#33992)
storybookjs/storybook: docs/ai/mcp/overview.mdx

<Callout variant="warning" icon="🧪">
  This is a [**preview**](../../releases/features.mdx#preview) feature and (though unlikely) the API may change in future releases. We welcome feedback and contributions to help improve this feature.
</Callout>

Storybook's MCP server connects your Storybook to AI agents, allowing them to understand your components and documentation, generate stories, run tests, and more. Agents will be equipped to re-use your existing components and follow your documented usage guidelines when generating UI, which helps ensure that the generated UI is consistent with your existing design system. Then they can write stories so you can preview the generated UI, and automatically run interaction tests (and accessibility checks) on those stories to validate their work. If any issues are found, the agent can fix them and re-run the tests to confirm that the issues are resolved, creating a self-healing loop that helps ensure the quality of the generated UI without requiring you to intervene.

## Installation

### 1. Install the addon

Run this command to install and register the Storybook MCP addon:

<CodeSnippets path="addon-mcp-add.md" />

> After restarting your Storybook, you can access the MCP server at `http://localhost:6006/mcp` (your port may be different). When viewed in the browser, you'll see a splash page that will redirect to the [manifest debugger](../manifests.mdx#debugging) after a few seconds.

Reviewer: I'm actually working on removing the automatic redirect and replacing it with a link here https://github.com/storybookjs/mcp/pull/179 because it was annoying and confusing UX. Unless you have strong opinions against, we should reword this once that PR is released.

### Development

The development toolset provides tools for authoring well-considered stories (including [interaction tests](../../writing-tests/interaction-testing.mdx)) and previewing those stories in the agent's chat interface. This allows you to easily check the agent's work, and the generated stories can serve as tests (which the agent can run, with the [testing toolset](#testing), to validate that the generated UI works as intended).

The development toolset includes the following tools:

#### `get-storybook-story-instructions`

Returns instructions to author useful stories, including which props to capture and how to write interaction tests for the story (if applicable).

#### `preview-stories`

Returns story previews rendered in the agent's chat interface, if the agent supports [MCP Apps](https://modelcontextprotocol.io/extensions/apps/overview#client-support). Otherwise, it returns links to relevant stories in Storybook.

<If renderer={['react']}>

### Docs

> The docs toolset provides tools that help the agent to re-use the components in your Storybook when generating UI. The agent can query the manifests to find components that match the requirements of the UI it's trying to build, and reference your documentation to understand how to use those components correctly. This helps ensure that the generated UI is consistent with your existing design system and makes use of the components you've already built.

Reviewer 2: Minor item here, but we can go with something as follows for clarity:
```suggestion
The docs toolset provides tools that help the agent reuse components from your Storybook when generating UI. The agent can query the manifests to find components that match the requirements of the UI it's trying to build, and reference your documentation to understand how to use those components correctly. This helps ensure the generated UI is consistent with your existing design system and uses the components you've already built.
```

The docs toolset includes the following tools:

#### `get-documentation`

Returns the detailed documentation for a specific component, including its props, the first three stories, an index of the remaining stories, and any additional documentation you've provided.

#### `get-documentation-for-story`

Returns the full story and associated documentation for a specific story. The agent calls this tool when `get-documentation` doesn't provide enough information for the agent to use a component effectively.

#### `list-all-documentation`

Returns an index containing components and unattached docs entries.

</If>

### Testing

> If you have [Storybook Test](../../writing-tests/index.mdx) set up in your Storybook, the testing toolset provides tools that allow the agent to run component tests (including [accessibility checks](../../writing-tests/accessibility-testing.mdx), if set up) and interpret their results. It also includes instructions to resolve issues. As the agent develops, it can choose to run tests to validate its work. If it finds issues, it can fix them and re-run the tests to confirm that the issues are resolved. This creates a _self-healing_ loop that helps ensure the quality of the generated UI, without requiring you to intervene.

Reviewer 2: The words are not precious but we can simplify this a bit:
```suggestion
If you have [Storybook Test](../../writing-tests/index.mdx) set up in your Storybook, the testing toolset provides tools that allow the agent to run component tests (including [accessibility checks](../../writing-tests/accessibility-testing.mdx), if set up) and interpret their results. It also includes instructions to resolve issues. As the agent develops, it can choose to run tests to validate its work. If it finds issues, it can fix them and re-run the tests to confirm they are resolved. This creates a _self-healing_ loop that helps ensure the quality of the generated UI, without requiring you to intervene.
```
